core/src/main/scala/io/odin/formatter/Formatter.scala

```diff
@@ -127,4 +154,18 @@ object Formatter {
         writeStackTrace(b, elements.tail, indent)
     }
   }
+
+  private def abbreviate(enclosure: String): String = {
+    @tailrec
+    def loop(input: List[String], builder: StringBuilder): StringBuilder = {
+      input match {
+        case Nil          => builder
+        case head :: Nil  => builder.append(head)
+        case head :: tail => loop(tail, builder.append(head.headOption.getOrElse('?')).append('.'))
```

Comment:
You can save an allocation of `Option` object by using a simple condition `if (head.isEmpty) "?" else head.head`.